### DISCUSSION TITLE
How can I handle errors what I can from database

### DISCUSSION THREAD

**Discussion by petar-arandjic** [ent/ent#2284] [Q&A] [answered]
So I'm developing a API using Go together with ORM package Entgo and Postgres. Is there a way to handle the errors same or similar way as they are solved in pg package where I have all needed information why query failed?

All I found in ent go are this methods but they only return bool
````
ent.IsNotFound(err)
ent.IsConstraintError(err)
ent.IsNotLoaded(err)
ent.IsValidationError(err)
````

It would be great to have something similar like pg.Error:
````
type Error struct {
  Severity         string
  Code             ErrorCode
  Message          string
  Detail           string
  Hint             string
  Position         string
  InternalPosition string
  InternalQuery    string
  Where            string
  Schema           string
  Table            string
  Column           string
  DataTypeName     string
  Constraint       string
  File             string
  Line             string
  Routine          string
}
````

**Comment by a8m** [comment, marked as answer]
Hey @petar-arandjic! ent wraps or returns the underlying driver errors. So, you can either use type-assertion or `errors.Is`/`errors.As` helpers to extract these types. See examples in our integration-test: [1](https://github.com/ent/ent/blob/60e0328/entc/integration/integration_test.go#L1845-L1858), [2](https://github.com/ent/ent/blob/60e0328/entc/integration/integration_test.go#L423).

**Reply by petar-arandjic** [reply]
Thank you for your help @a8m

**Reply by shinebayar-g** [reply]
Is there a common place for the list of error definitions? I'm trying to find how to check for `ent: x not found` error type.

Update: Ah got it, at least in my case it's like `ent.IsNotFound(err)`
